.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,24 @@
+# Release Description
+<!--- Summarize what is being released.  -->
+
+# Release Checklist
+<!--- Fill out the following Release checklist -->
+- [ ] Version number has been updated
+- [ ] Description file updated with New Developers (if applicable)
+- [ ] NEWS.md has been updated and issues linked where appropriate
```

Comment:
Do we need this? I thought we moved this to the feature PR review because it is hard to check at release time if all updates are in NEWS.md.

Reply:
I guess I was thinking of a final spot check of the `News.md` for links and format.  Just a big update and no one has been assigned to go over the entire changelog.  Was thinking this was a good opportunity to review it before it goes live.